addOns/quickstart/src/main/java/org/zaproxy/zap/extension/quickstart/ajaxspider/AjaxSpiderExplorer.java

```diff
@@ -0,0 +1,163 @@
+import org.zaproxy.zap.view.LayoutHelper;
+
+public class AjaxSpiderExplorer implements PlugableSpider {
+
+    private ExtensionQuickStartAjaxSpider extension;
+    private JCheckBox selectCheckBox;
+    private JComboBox<ProvidedBrowserUI> browserComboBox;
+
+    public AjaxSpiderExplorer(ExtensionQuickStartAjaxSpider extension) {
+        this.extension = extension;
+    }
+
+    public ExtensionAjax getExtAjax() {
+        return Control.getSingleton().getExtensionLoader().getExtension(ExtensionAjax.class);
+    }
+
+    @Override
+    public void startScan(URI uri) {
+        ExtensionAjax extAjax = this.getExtAjax();
+        AjaxSpiderParam options = new AjaxSpiderParam();
+        options.load(new ZapXmlConfiguration());
+        ProvidedBrowserUI browserUi = (ProvidedBrowserUI) getBrowserComboBox().getSelectedItem();
+        options.setBrowserId(browserUi.getBrowser().getId());
+        AjaxSpiderTarget.Builder builder = AjaxSpiderTarget.newBuilder(extension.getModel().getSession());
+        builder.setStartUri(uri);
+        builder.setInScopeOnly(false);
+        builder.setSubtreeOnly(false);
+        builder.setOptions(options);
+        extAjax.startScan(builder.build());
+    }
+
+    @Override
+    public void stopScan() {
+        this.getExtAjax().stopScan();
+    }
+
+    @Override
+    public String getLabel() {
+        return Constant.messages.getString("quickstart.label.ajaxspider");
+    }
+
+    private JCheckBox getSelectCheckBox() {
+        if (selectCheckBox == null) {
+            selectCheckBox = new JCheckBox();
+            selectCheckBox.addActionListener(new ActionListener() {
+                @Override
+                public void actionPerformed(ActionEvent arg0) {
+                    getBrowserComboBox().setEnabled(selectCheckBox.isSelected());
+                    extension.getExtQuickStart().getQuickStartParam().setAjaxSpiderEnabled(selectCheckBox.isSelected());
+                }
+            });
+        }
+        return selectCheckBox;
+    }
+
+    private JComboBox<ProvidedBrowserUI> getBrowserComboBox() {
+        if (browserComboBox == null) {
+            browserComboBox = new JComboBox<ProvidedBrowserUI>();
+            ProvidedBrowsersComboBoxModel model = extension.getExtSelenium().createProvidedBrowsersComboBoxModel();
+            model.setIncludeHeadless(false);
+            model.setIncludeUnconfigured(false);
+            browserComboBox.setModel(model);
+            browserComboBox.addActionListener(new ActionListener() {
+
+                @Override
+                public void actionPerformed(ActionEvent arg0) {
+                    extension.getExtQuickStart().getQuickStartParam()
+                            .setAjaxSpiderDefaultBrowser(browserComboBox.getSelectedItem().toString());
+                }
+            });
+        }
+        return browserComboBox;
+    }
+
+    @Override
+    public JPanel getPanel() {
+        JPanel ajaxSpiderPanel = new JPanel(new GridBagLayout());
```

Comment:
This needs to return the same panel always, otherwise the removal does not work properly (as it tries to remove a different panel).